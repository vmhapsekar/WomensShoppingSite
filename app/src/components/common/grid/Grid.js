import React from 'react'

import {ThemeProvider} from 'styled-components'
import {Row, Col} from 'react-styled-flexboxgrid'

import {theme, StyledGrid} from './styles'

// Defining Grid to make it responsive
export const Grid = (props) => (
  <ThemeProvider theme={theme}>
    <StyledGrid {...props}>
      {props.children}
    </StyledGrid>
  </ThemeProvider>
)

// Defining Row of the Grid
export const GRow = (props) => (
  <ThemeProvider theme={theme}>
    <Row {...props}>
      {props.children}
    </Row>
  </ThemeProvider>
)

// Defining Column of the Grid
export const GCol = (props) => (
  <ThemeProvider theme={theme}>
    <Col {...props}>
      {props.children}
    </Col>
  </ThemeProvider>
)
