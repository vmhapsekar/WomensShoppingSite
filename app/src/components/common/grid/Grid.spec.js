import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import {Grid, GRow, GCol} from './Grid.js'

describe('<Grid />', () => {
  it('WILL match snapshot: <Grid />', () => {
    const wrapper = renderer.create(<Grid fluid />)
    expect(wrapper).toMatchSnapshot()
  })
  it('WILL match snapshot: <GRow />', () => {
    const wrapper = renderer.create(<GRow />)
    expect(wrapper).toMatchSnapshot()
  })
  it('WILL match snapshot: <GCol />', () => {
    const wrapper = renderer.create(<GCol xs={12} sm={6} md={4} lg={3}>Text</GCol>)
    expect(wrapper).toMatchSnapshot()
  })
})
