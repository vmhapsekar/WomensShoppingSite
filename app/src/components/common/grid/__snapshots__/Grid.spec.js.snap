// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Grid /> WILL match snapshot: <GCol /> 1`] = `
.c0 {
  box-sizing: border-box;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  padding-right: 0.9375rem;
  padding-left: 0.9375rem;
}

@media only screen and (min-width:0em) {
  .c0 {
    -webkit-flex-basis: 100%;
    -ms-flex-preferred-size: 100%;
    flex-basis: 100%;
    max-width: 100%;
    display: block;
  }
}

@media only screen and (min-width:30em) {
  .c0 {
    -webkit-flex-basis: 50%;
    -ms-flex-preferred-size: 50%;
    flex-basis: 50%;
    max-width: 50%;
    display: block;
  }
}

@media only screen and (min-width:48em) {
  .c0 {
    -webkit-flex-basis: 33.333333333333336%;
    -ms-flex-preferred-size: 33.333333333333336%;
    flex-basis: 33.333333333333336%;
    max-width: 33.333333333333336%;
    display: block;
  }
}

@media only screen and (min-width:60em) {
  .c0 {
    -webkit-flex-basis: 25%;
    -ms-flex-preferred-size: 25%;
    flex-basis: 25%;
    max-width: 25%;
    display: block;
  }
}

<div
  className="c0"
>
  Text
</div>
`;

exports[`<Grid /> WILL match snapshot: <GRow /> 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-right: -0.9375rem;
  margin-left: -0.9375rem;
}

<div
  className="c0"
/>
`;

exports[`<Grid /> WILL match snapshot: <Grid /> 1`] = `
.c1 {
  margin-right: auto;
  margin-left: auto;
  padding-right: 0.9375rem;
  padding-left: 0.9375rem;
}

.c0 {
  max-width: 1164px;
  margin: 0 auto;
}

<div
  className="c0 c1"
/>
`;
